app/models/user.rb

```diff
@@ -22,12 +31,23 @@ def to_s
   # @param [OmniAuth::AuthHash] auth
   def self.from_omniauth(auth)
     Rails.logger.debug "auth = #{auth.inspect}"
-    # Uncomment the debugger above to capture what a shib auth object looks like for testing
-    user = where(provider: auth.provider, uid: auth.info.uid).first_or_create
-    user.display_name = auth.info.display_name
-    user.uid = auth.info.uid
-    user.email = auth.info.mail
+    raise User::NilShibbolethUserError.new("No uid", auth) if auth.uid.empty? || auth.info.uid.empty?
+    user = User.find_or_initialize_by(provider: auth.provider, uid: auth.info.uid)
+    user.assign_attributes(display_name: auth.info.display_name)
+    # tezprox@emory.edu isn't a real email address
+    user.email = auth.info.uid + '@emory.edu' unless auth.info.uid == 'tezprox'
     user.save
     user
+  rescue User::NilShibbolethUserError => e
+    Rails.logger.error "Nil user detected: Shibboleth didn't pass a uid for #{e.auth.inspect}"
+  end
+
+  def self.log_omniauth_error(auth)
```

Comment:
Are we using this method in this class anywhere or any other class? If not, does it make sense to remove it? I see we already have error handling above.

Reply:
Yeah, good point. I'll remove it.